llvm/docs/RFCProcess.rst

```diff
@@ -0,0 +1,80 @@
+=================================
+Request For Comment (RFC) process
+=================================
+
+.. contents::
+   :local:
+   :depth: 1
+
+Introduction
+============
+Substantive changes to LLVM projects need to be acceptable to the wider
+community, which requires gaining community consensus to adopt the changes.
+This is done by posting an RFC and obtaining feedback about the proposal.
+
+Process
+=======
+
+Writing an RFC
+--------------
+The process begins with writing a proposal for the changes you'd like to see
+made. The proposal should include:
+
+* a high-level overview of what changes are being proposed,
+* information detailing the motivation for why the changes are being proposed,
+* detailed information the proposed changes and how they impact different parts
+  of the project, and
+* a list of any open questions the community should explicitly address.
+
+Once the contents of the proposal are ready, the proposal should be posted to
+the appropriate forum on `Discourse <https://discourse.llvm.org/>`_.
+
+Feedback Period
+---------------
+Once the RFC is posted, the community will provide feedback on the proposal.
+The feedback period is a collaborative effort between the community and the
+proposal authors. Authors should take the community's feedback into
+consideration and edit the original post to incorporate relevant changes they
+agree to. Edits should be made such that it's clear what has changed. Editing
+the original post makes it easier for the community to understand the proposal
+without having to read every comment on the thread, though this can make
+reading the comment thread somewhat more difficult as comments may be referring
+to words no longer in the proposal.
+
+There is not a set time limit to the feedback period; it lasts as long as
+discussion is actively continuing on the proposal.
+
+Trivial Acceptance or Rejection
+-------------------------------
+If the proposal has obvious consensus (for or against), a maintainer for each
+of the impacted parts of the project will explicitly accept or reject the RFC
+by leaving a comment stating their decision and possibly detailing any
+provisions for their acceptance. Additionally, the original post should be
```

Comment:
This seems good, though I'm not sure if it reflects existing practice? And it's a bit awkward requiring RFC authors to do that - I imagine some might not know/remember to do that, or feel a bit bad about being asked to, themselves, update their RFC to say it was rejected (feels a bit like rubbing salt in the wound)... not sure how to do that better, though.

Reply:
Yeah, I think this only really works if the accepting maintainer also has edit rights on Discourse (which have to be granted by admins). @AaronBallman has those permissions, so he can edit in the acceptance for clang RFCs, but most other maintainers won't be able to do that.

Reply:
Yeah, it's something I've been doing in Clang for a while now at the request of community members, but the fact that Discourse requires special privileges to do it means this probably should be walked back. I'll update accordingly.

Reply:
Done